### PR TITLE
Unit tests for virtual grants

### PR DESCRIPTION
**Pull request by masparrow**
Resolves https://github.com/brave/brave-browser/issues/6949

- Bind Utils
- Promotion
- Promotion Utils
- Security Helper (Ledger

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

Nb. I have two failing unit tests which are unrelated to my changes, and they seem to be failing on master - otherwise all unit tests pass - (888/890); 

```
2 tests crashed:
    MediaYouTubeTest.GetChannelUrl (../../brave/vendor/bat-native-ledger/src/bat/ledger/internal/media/youtube_unittest.cc:498)
    MediaYouTubeTest.GetVideoUrl (../../brave/vendor/bat-native-ledger/src/bat/ledger/internal/media/youtube_unittest.cc:466
```

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat/ledger/internal/common/bind_util.cc [hunk 1]
```diff
@@ -109,10 +161,11 @@ std::string FromPromotionToString(const ledger::PromotionPtr info) {
   promotion.SetStringKey("approximate_value",
       std::to_string(info->approximate_value));
   promotion.SetStringKey("expires_at", std::to_string(info->expires_at));
+  promotion.SetStringKey("claimed_at", std::to_string(info->claimed_at));
```
Comment (masparrow) [inline vendor/bat-native-ledger/src/bat/ledger/internal/common/bind_util.cc]: Discovered as missing, by the new unit tests for this file.

File: vendor/bat-native-ledger/src/bat/ledger/internal/common/bind_util.cc [hunk 2]
```diff
@@ -109,10 +161,11 @@ std::string FromPromotionToString(const ledger::PromotionPtr info) {
...
   promotion.SetIntKey("version", info->version);
   promotion.SetIntKey("type", static_cast<int>(info->type));
   promotion.SetIntKey("suggestions", info->suggestions);
-  promotion.SetBoolKey("status", static_cast<int>(info->status));
+  promotion.SetIntKey("status", static_cast<int>(info->status));
```
Comment (masparrow) [inline vendor/bat-native-ledger/src/bat/ledger/internal/common/bind_util.cc]: Discovered as using the wrong `Set...Key` by the unit test for this file.

File: vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion.cc [hunk 3]
```diff
@@ -545,6 +545,7 @@ void Promotion::OnFetchSignedTokens(
   }
 
   std::string json_creds;
+  // TODO(masparrow): this derefences pointers without guarding
```
Comment (masparrow) [inline vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion.cc]: Will convert to a ticket / capture in an existing ticket.

File: vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion.h [hunk 4]
```diff
@@ -47,6 +47,13 @@ class Promotion {
       ledger::PromotionPtr promotion,
       ledger::ResultCallback callback);
 
+  void SetLastCheckTimerIdForTesting(uint32_t timer_id) {
+      last_check_timer_id_ = timer_id;
+    }
+  void SetRetryTimerIdForTesting(uint32_t timer_id) {
+      retry_timer_id_ = timer_id;
+    }
+
```
Comment (masparrow) [inline vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion.h]: Will implement in the .cc

File: vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc [hunk 5]
```diff
@@ -5,47 +5,243 @@
...
+  promotion_->SetLastCheckTimerIdForTesting(0);
+
+  // Act
+  promotion_->Refresh(true);
+}
+
+// void OnTimer(const uint32_t timer_id);
+// Nb. Cannot test without mocking Promotion
+//   discussed with tmancey and no value as code may be changing
+//   need to confirm with nejc
```
Comment (masparrow) [inline vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc]: @NejcZdovc Thoughts here?

File: components/definitions/rewards.d.ts [hunk 6]
```diff
@@ -116,11 +116,13 @@ declare namespace Rewards {
   export type CaptchaStatus = 'start' | 'wrongPosition' | 'generalError' | 'finished' | null
 
   export enum PromotionTypes {
+    UNKNOWN = -1,
```
Comment (masparrow) [inline components/definitions/rewards.d.ts]: I'd rather use `UNKNOWN = 0` but `0` is in use and is persisted to disk I believe? 

`UNKNOWN` is needed (or perhaps it should be `INVALID`) as values can be parsed from arbitrary int's in JSON from the server, so if illegal values are sent we drop to UNKNOWN in code below.

File: vendor/bat-native-ledger/src/bat/ledger/internal/common/bind_util.cc [hunk 7]
```diff
@@ -12,6 +12,58 @@
...
+    case ledger::PromotionStatus::OVER: {
+      break;
+    }
+    default: {
+      promotion_status = ledger::PromotionStatus::UNKNOWN;
+    }
+  }
+  return promotion_status;
+}
+
```
Comment (masparrow) [inline vendor/bat-native-ledger/src/bat/ledger/internal/common/bind_util.cc]: These 'ConvertIntTo' enum helpers might be better as more accessible methods somewhere else?

File: vendor/bat-native-ledger/src/bat/ledger/internal/common/bind_util.cc [hunk 8]
```diff
@@ -155,14 +208,19 @@ ledger::PromotionPtr FromStringToPromotion(const std::string& data) {
     promotion->expires_at = std::stoull(*expires_at);
   }
 
+  auto* claimed_at = dictionary->FindStringKey("claimed_at");
+  if (claimed_at) {
+    promotion->claimed_at = std::stoull(*claimed_at);
+  }
+
```
Comment (masparrow) [inline vendor/bat-native-ledger/src/bat/ledger/internal/common/bind_util.cc]: Discovered as missing, by the new unit tests for this file.

File: vendor/bat-native-ledger/src/bat/ledger/internal/common/bind_util.cc [hunk 9]
```diff
@@ -57,7 +109,7 @@ ledger::ContributionQueuePtr FromStringToContributionQueue(
 
   auto* type = dictionary->FindKey("type");
   if (type && type->is_int()) {
-    queue->type = static_cast<ledger::RewardsType>(type->GetInt());
+    queue->type = ConvertIntToRewardsType(type->GetInt());
```
Comment (masparrow) [inline vendor/bat-native-ledger/src/bat/ledger/internal/common/bind_util.cc]: The new unit tests pass in 'illegal' values for enum's (i.e. RewardsType) in sample JSON - and they were being cast to 'illegal' enum values which would trip up logic elsewhere. 

Instead we should be validating all enumerated types received from the server as `int` and handling accordingly. I'll raise a ticket to make sure code which uses these three enum types deals with `UNKNOWN` values.

File: vendor/bat-native-ledger/include/bat/ledger/public/interfaces/ledger.mojom [hunk 10]
```diff
@@ -246,6 +246,7 @@ struct TransferFee {
 };
 
 enum RewardsType {
+  UNKNOWN = -1,
   AUTO_CONTRIBUTE = 2,
```
Comment (tmancey) [inline vendor/bat-native-ledger/include/bat/ledger/public/interfaces/ledger.mojom]: Not to do with this PR, but why are we bit-shifting these values?

File: vendor/bat-native-ledger/src/bat/ledger/internal/common/bind_util.cc [hunk 11]
```diff
@@ -12,6 +12,58 @@
...
+  ledger::RewardsType rewards_type = static_cast<ledger::RewardsType>(type);
+
+  switch (rewards_type) {
+    case ledger::RewardsType::AUTO_CONTRIBUTE:
+    case ledger::RewardsType::ONE_TIME_TIP:
+    case ledger::RewardsType::RECURRING_TIP: {
+      break;
+    }
+    default: {
+      rewards_type = ledger::RewardsType::UNKNOWN;
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/common/bind_util.cc]: I assume we have a default due to legacy code persisting other values?

File: vendor/bat-native-ledger/src/bat/ledger/internal/common/bind_util.cc [hunk 12]
```diff
@@ -12,6 +12,58 @@
...
+  ledger::PromotionType promotion_type =
+      static_cast<ledger::PromotionType>(type);
+
+  switch (promotion_type) {
+    case ledger::PromotionType::UGP:
+    case ledger::PromotionType::ADS: {
+      break;
+    }
+    default: {
+      promotion_type = ledger::PromotionType::UNKNOWN;
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/common/bind_util.cc]: I assume we have a default due to legacy code persisting other values?

File: vendor/bat-native-ledger/src/bat/ledger/internal/common/bind_util.cc [hunk 13]
```diff
@@ -12,6 +12,58 @@
...
+    case ledger::PromotionStatus::ACTIVE:
+    case ledger::PromotionStatus::ATTESTED:
+    case ledger::PromotionStatus::CLAIMED:
+    case ledger::PromotionStatus::SIGNED_TOKENS:
+    case ledger::PromotionStatus::FINISHED:
+    case ledger::PromotionStatus::OVER: {
+      break;
+    }
+    default: {
+      promotion_status = ledger::PromotionStatus::UNKNOWN;
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/common/bind_util.cc]: I assume we have a default due to legacy code persisting other values?

File: vendor/bat-native-ledger/src/bat/ledger/internal/common/bind_util.cc (hunk 9 above)
Comment (tmancey) [reply vendor/bat-native-ledger/src/bat/ledger/internal/common/bind_util.cc]: ah, you've answered my questions above :-)

File: vendor/bat-native-ledger/include/bat/ledger/public/interfaces/ledger.mojom (hunk 10 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/include/bat/ledger/public/interfaces/ledger.mojom]: @tmancey legacy code 😄 but because we have this values in the db when I migrated it mojo I kept the same values.

File: vendor/bat-native-ledger/src/bat/ledger/internal/common/bind_util.cc (hunk 9 above)
Comment (masparrow) [reply vendor/bat-native-ledger/src/bat/ledger/internal/common/bind_util.cc]: Raised https://github.com/brave/brave-browser/issues/7897 to cover the above

**Comment by NejcZdovc** [comment]
CI is failing on unit tests https://ci.brave.com/job/brave-browser-build-pr/job/issues%252F6949/1/execution/node/591/log/

**Comment by masparrow** [comment]
> CI is failing on unit tests https://ci.brave.com/job/brave-browser-build-pr/job/issues%252F6949/1/execution/node/591/log/

Most failing are unrelated, but these two; 

- PromotionTest.TestRefreshWithRetryWithTimerId
- PromotionTest.TestRefreshWithoutRetryWithTimerId

failed and are related, but passed locally. The only failing tests I had locally were the YouTube related Media ones. 

I'll check it all out again, and rebase if needed.

**Comment by tmancey** [comment]
CI for macOS failed for unrelated reasons, restarting macOS

File: vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc [hunk 14]
```diff
@@ -10,11 +10,10 @@
 #include <vector>
 
 #include "base/test/task_environment.h"
-#include "bat/ledger/internal/ledger_client_mock.h"
-#include "bat/ledger/internal/ledger_impl_mock.h"
 #include "bat/ledger/internal/promotion/promotion.h"
-#include "bat/ledger/ledger.h"
 #include "testing/gtest/include/gtest/gtest.h"
+#include "bat/ledger/internal/ledger_client_mock.h"
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc]: please keep imports as they were as they were sorted

File: vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc [hunk 15]
```diff
@@ -65,31 +64,242 @@ class PromotionTest : public testing::Test {
...
+  // Act
+  promotion_->ClaimTokens(promotion.Clone(), callback);
+
+  // Assert
+  EXPECT_FALSE(callback_called);
+}
+
+TEST_F(PromotionTest, LegacyPromotionIsNotOverwritten) {
+  const std::string payment_id = "this_is_id";
+  ON_CALL(*mock_ledger_impl_, GetPaymentId())
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc]: let's keep this in SetUp and just add variable with default value which can be overwritten

File: vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc [hunk 16]
```diff
@@ -65,31 +64,242 @@ class PromotionTest : public testing::Test {
...
       .WillByDefault(testing::ReturnRefOfCopy(payment_id));
 
-    const std::string wallet_passphrase = "phrase";
-    ON_CALL(*mock_ledger_impl_, GetWalletPassphrase())
+  const std::string wallet_passphrase = "phrase";
+  ON_CALL(*mock_ledger_impl_, GetWalletPassphrase())
       .WillByDefault(testing::Return(wallet_passphrase));
 
-    ON_CALL(*mock_ledger_impl_, LoadURL(_, _, _, _, _, _))
+  ON_CALL(*mock_ledger_impl_, LoadURL(_, _, _, _, _, _))
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc]: why was this moved out of SetUp?

File: vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc (hunk 15 above)
Comment (tmancey) [reply vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc]: `GetPaymentId()` is only needed for this test, why add for all tests?

File: vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc [hunk 17]
```diff
@@ -65,31 +64,242 @@ class PromotionTest : public testing::Test {
...
+  // Act
+  promotion_->ClaimTokens(nullptr, callback);
+
+  // Assert
+  EXPECT_TRUE(callback_called);
+}
+
+TEST_F(PromotionTest, TestClaimTokensWithPromotion) {
+  // Arrange
+//  EXPECT_CALL(*mock_ledger_impl_, LoadURL(_, _, _, _, _, _));
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc]: Why commented out?

File: vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc (hunk 16 above)
Comment (tmancey) [reply vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc]: As above this is only needed for a single test?

File: vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc (hunk 15 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc]: that is not true, you can see that @masparrow uses it in multiple places

File: vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc (hunk 16 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc]: so that when you need LoadUrl, you already have flow ready you just add response in one array and you are good to go

File: vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc (hunk 15 above)
Comment (tmancey) [reply vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc]: He does but with different fake payment id's?

Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc]: that's why I said variable

Comment (tmancey) [reply vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc]: It is used in multiple places but not all, and to add a global variable for me does not make sense

Comment (masparrow) [reply vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc]: Having a global Setup method isn't viable for multiple tests - we should only provide global Setup's when they are truly global, otherwise we should set up per test which is why I moved the code.

File: vendor/bat-native-ledger/src/bat/ledger/internal/test/mock_ledger_client.cc [hunk 18]
```diff
@@ -55,13 +55,4 @@ void MockLedgerClient::SavePublisherState(
...
 
-void MockLedgerClient::LoadURL(const std::string& url,
-                               const std::vector<std::string>& headers,
-                               const std::string& content,
-                               const std::string& contentType,
-                               const ledger::UrlMethod method,
-                               ledger::LoadURLCallback callback) {
-  callback(true, "{}", {});
-}
-
```
Comment (masparrow) [inline vendor/bat-native-ledger/src/bat/ledger/internal/test/mock_ledger_client.cc]: This mock was providing a 'default' mock logic which made no sense, as it prevents mocking it. I've repurposed it to a true MOCK method, and all unit tests pass (including mine which were failing due to this).

File: vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_unittest.cc [hunk 19]
```diff
@@ -39,7 +39,7 @@ class UpholdTest : public testing::Test {
 };
 
 TEST_F(UpholdTest, FetchBalanceConnectedWallet) {
-  EXPECT_CALL(*mock_ledger_impl_, LoadURL(_, _, _, _, _, _)).Times(0);
+  EXPECT_CALL(*mock_ledger_client_, LoadURL(_, _, _, _, _, _)).Times(0);
```
Comment (masparrow) [inline vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_unittest.cc]: So here, because this is not a public member of LedgerImpl, we have to mock the client level, hence the changes above (otherwise this test - which was originally passing - fails).

Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_unittest.cc]: `LoadURL` is public method in ledger_impl.h https://github.com/brave/brave-core/blob/ab783ba8688d225aac90dd4847247958a41c5ab3/vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.h#L229

Comment (masparrow) [reply vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_unittest.cc]: 😕 clearly having a brain-melt there! 

However, the LoadURL changes are required.

File: vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc (hunk 14 above)
Comment (masparrow) [reply vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc]: Sorted. Literally. :+1:

**Comment by masparrow** [comment]
Travis fail is due to : TEST_SUITE=test-security - unrelated to this PR

Jenkins fail is due to it skipping Mac OS when Mac OS is not set for skip - so will rebase and redo a build without any CI skip flags.

File: vendor/bat-native-ledger/src/bat/ledger/internal/test/mock_ledger_client.h [hunk 20]
```diff
@@ -29,6 +29,14 @@ class MockLedgerClient : public ledger::LedgerClient {
 
   void CreateWallet(ledger::LedgerCallbackHandler* handler) override;
 
+  MOCK_METHOD6(LoadURL,
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/test/mock_ledger_client.h]: why was this added here? we have `ledger_client_mock.h` where we define mocks

File: vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_unittest.cc (hunk 19 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_unittest.cc]: still not sure that I understand this change. It goes through `mock_ledger_impl_`, something else should be going on

File: vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc [hunk 21]
```diff
@@ -65,31 +65,259 @@ class PromotionTest : public testing::Test {
...
   }
+};
+
+TEST_F(PromotionTest, TestInitialize) {
+  // Arrange
+  ON_CALL(*mock_ledger_client_, GetBooleanState(
+    ledger::kStatePromotionCorruptedMigrated))
+     .WillByDefault(testing::Return(false));
+
+  EXPECT_CALL(*mock_ledger_impl_, GetAllPromotions(_)).Times(2);
```
Comment (masparrow) [inline vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc]: @NejcZdovc `promotion->Initialize()` invokes the `GetAllPromotions` callback twice in this scenario, once to perform the `CheckForCorrupted` and then it falls through to invoke `Retry`... is this the expected behaviour? As it means `CheckForCorrupted` is invoking an async `LoadURL`, and then the code moves on to processing local data in `Retry`

File: vendor/bat-native-ledger/src/bat/ledger/internal/test/mock_ledger_client.h (hunk 20 above)
Comment (masparrow) [reply vendor/bat-native-ledger/src/bat/ledger/internal/test/mock_ledger_client.h]: Ah, that's confusing naming then :D - but, this appear to be the 'mock' in use by the code, as this change resolved the failing tests (related to the discussion we had on zoom).

File: vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_unittest.cc (hunk 19 above)
Comment (masparrow) [reply vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_unittest.cc]: Happy to change - though the changes were needed due to the call paths being invoked by the existing code, which I/we shouldn't be changing to allow a unit test to pass (of course). So the mock itself had to be changed to accommodate the code. If this can be done in a different / better way - I'm all ears 👍

File: vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc (hunk 21 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc]: we could make it sync just for double safety

Comment (masparrow) [reply vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc]: @NejcZdovc do you want me to make that change as part of this PR? (I'd say it's a little out of scope tbh and is better as a new issue IMHO).

File: vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_unittest.cc (hunk 19 above)
Comment (masparrow) [reply vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_unittest.cc]: - [ ] Ticket for the mock issues discussed in this PR

File: vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc (hunk 21 above)
Comment (masparrow) [reply vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc]: - [ ] Ticket for GetAllPromotions rethink

Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc]: new issue is fine
